tests/50federation/30room-join.pl

```diff
@@ -327,6 +326,227 @@ sub assert_is_valid_pdu {
    };
 
 
+
+test "Inbound federation supports v2 /send_join",
```

Comment:
Can we not reuse the `Inbound federation can receive v1 room-join requests` test here? I.e. something like:

```perl
foreach my $version ( qw( v1 v2 ) ) {
   test "Inbound federation supports $version /send_join",
      ...
}
```

or something?